include/experimental/__p0009_bits/layout_stride.hpp

```diff
@@ -309,6 +315,44 @@ struct layout_stride {
        */
     }
 
+    MDSPAN_TEMPLATE_REQUIRES(
+      class IntegralTypes,
+      /* requires */ (
+        // MSVC 19.32 does not like using index_type here, requires the typename Extents::index_type
+        // error C2641: cannot deduce template arguments for 'MDSPAN_IMPL_STANDARD_NAMESPACE::layout_stride::mapping'
+        _MDSPAN_TRAIT(std::is_convertible, const std::remove_const_t<IntegralTypes>&, typename Extents::index_type) &&
+        _MDSPAN_TRAIT(std::is_nothrow_constructible, typename Extents::index_type, const std::remove_const_t<IntegralTypes>&)
+      )
+    )
+    MDSPAN_INLINE_FUNCTION
+    constexpr
+    mapping(
+      mdspan_non_standard_tag,
```

Comment:
I appreciate the tag!  This will make it clear that the feature is nonstandard.

Our approach to issues like this has been to create our own copies of Standard classes and make them available in libcu++.  For example, `#include <cuda/std/array>` gets you `cuda::std::array`, which acts just like `std::array`, but has the appropriate `__device__ __host__` blessings.  On the other hand, this approach is fine, especially since the tag makes it hard to call this by accident.

Reply:
Haha yeah Christian came up with the tag idea.

This change was necessary because of a particular use case in Kokkos proper and we can't use span yet. Maybe we should include an "array" class in mdspan in the future but at least for now I just wanted to get something that was usable since it was blocking a Kokkos feature.